samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -364,4 +365,26 @@ public void run(Duration timeout) {
         .collect(Collectors.toMap(entry -> entry.getKey().getPartition().getPartitionId(),
             entry -> entry.getValue().stream().map(e -> (T) e.getMessage()).collect(Collectors.toList())));
   }
+
+  private TaskFactory createTaskFactory() {
```

Comment:
Instead of adapting to the legacy task class, would it be possible to just pass the TaskApplication directly through a constructor?

Reply:
There is a wide-spread usage of TestRunner.of(Class taskClass) in the test code. I would prefer to address that in a separate PR. Opened SAMZA-1818 to track this issue.